### PR TITLE
docs: VRG usage

### PR DESCRIPTION
**Pull request by hatfieldbrian**
This PR documents:
- Application protection, unprotection, failover, and failback using the VRG interface
- Kubernetes resource protection

### CODE REVIEW

File: docs/krp.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                - backupName: everything
+                    includeClusterResources: true
+                    excludedResources: ["ConfigMap", "Secret", "Deployment", "sample1.myapp.mycompany.com", "sample2.myapp.mycompany.com", "sample3.myapp.mycompany.com"]  # don't restore resources we've already restored
+```
+
+## Explanation of the Capture and Recovery Specifications in the VRG
+
+The scope of a VRG disaster protection is a single Kubernetes namespace.  The
+VRG protects persistent volumes associated with the namespace and optionally
+protects Kubernetes resources in the namespace.  This documentation covers the
```
Comment (hatfieldbrian) [inline docs/krp.md:86]: (nit) consider appending "and/or at cluster scope" to the end of the sentence

File: docs/krp.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+1. The name of each item in the captureOrder list must be unique
+1. The backupName of each item in the recoverOrder list much match a name in the
+ recoverOrder list
+1. A labelSelector in a list item only applies to that item in the list
+1. If a list item contains multiple labelSelectors then any resource that
+ matches either label selector is operated upon
+1. includeClusterResources in a list item only applies to that item in the list
+1. Each list item can contain either an includedResources section or an
+ excludedResources section, but not both
```
Comment (hatfieldbrian) [inline docs/krp.md:117]: I'm not sure this is true.  I think both can be specified.  But the only case that comes to mind for it is including resources matching a substring (e.g., *.apps) and then excluding some of those matches (e.g., replicasets.apps).  I haven't tested this and don't know if velero supports substring matching.

File: docs/krp.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+attempted, achieved, and maintained.  It is up to the application to deal with
+asynchronous behavior required by this set and attempt, achieve, maintain
+architecture.  However, in highly complex applications with stringent user
+expectations the level of asynchrony can become unmanageable.  First, the scope
+of asynchrony can break the ability of programmers to anticipate all sequences
+of events.  Second, the scope of asynchrony can be the source of failed
+dependencies resulting in backoff retry loops which can violate the application
+Recovery Time Objective (RTO).  Restoring resources in a prescribed order can
+avoid both of these problems.  So the Ramen VRG provides a mechanism to support
+capturing and restoring resources in a proscribed order.
```
Comment (hatfieldbrian) [inline docs/krp.md:46]: (spelling) prescribed

**Review by hatfieldbrian — COMMENTED** [review]
@pmuench1717 as a maintainer I think you may be able to commit changes or you can just send me a file.  I can squash the commits down to one before merging

File: docs/krp.md (hunk 2 above)
Comment (pmuench1717) [reply docs/krp.md:117]: What the code does and what we want users to know is different in this case.  So I'm documenting what we want users to know.

File: docs/krp.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                    includedResources: ["ConfigMap", "Secret"]
+                - name: custom
+                    includedResources: ["sample1.myapp.mycompany.com", "sample.myapp.mycompany.com", "sample3.myapp.mycompany.com"]
+                    labelSelector: "myAppPersist"
+                    # includeClusterResources: false # by default
+                - name: deployments
+                    includedResources: ["Deployment"]
+                - name: everything
+                    includeClusterResources: true
+                    excludedResources: [""]  # include everything with no history, even resources in other backups
```
[the remark this answers is not in the document]
Comment (pmuench1717) [reply docs/krp.md:66]: I'm OK with leaving it explicit for now.  We can change later if users complain.

File: docs/krp.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+Recovery Time Objective (RTO).  Restoring resources in a prescribed order can
+avoid both of these problems.  So the Ramen VRG provides a mechanism to support
+capturing and restoring resources in a proscribed order.
+
+## An Example Kubernetes Resource Protection Specification
+
+```yaml
+    spec:
+        kubeObjectProtection:
+            # backup section
```
[the remark this answers is not in the document]
Comment (pmuench1717) [reply docs/krp.md:53]: The current approach is inconsistent.  From the users point of view he sees the term "capture" once for "captureOrder", but he sees the term "backup" each time he uses "backupName".  So from the user point of view it seems more inconsistent to talk about capturing than backing up.  I like your terms better and would like them to be used consistently across the user interface, but given the inconsistency I'm not convinced either approach is all that better.

**Review by hatfieldbrian — COMMENTED** [review]
Looks good to me

**Pull request merged**